lib/src/slices/api_auth_service/api_auth_service.dart

```diff
@@ -103,4 +116,64 @@ class ApiAuthService {
       ApiAuthServiceAccountModel apiAuthServiceAccountModel) async {
     await _apiAuthServiceRepository.delete(apiAuthServiceAccountModel);
   }
+
+  Future<void> signOutAccounts() async {
+    List<ApiAuthServiceAccountModel> accounts = await this.getAllAccounts();
+    accounts.forEach((account) async {
+      ApiAuthServiceProviderInterface? provider = getProvider(account);
+      if (provider != null) {
+        provider.logOut(account);
+      }
+    });
+  }
+
+  Future<List<ApiAuthServiceAccountModel>> getAllAccounts() async {
+    return await _apiAuthServiceRepository.getAll();
+  }
+
+  Future<ApiAuthServiceAccountModel?> getAccountById(int accountId) async {
+    return await _apiAuthServiceRepository.getById(accountId);
+  }
+
+  Future<ApiAuthServiceAccountModel?> linkAccount(String providerName) async {
+    ApiAuthServiceAccountModel? account;
+    AuthorizationTokenResponse? tokenResponse =
+        await this.authorizeAndExchangeCode(providerName: providerName);
+    if (tokenResponse != null) {
+      ApiAuthServiceAccountModel apiAuthServiceAccountModel =
+          ApiAuthServiceAccountModel(
+              provider: providerName,
+              accessToken: tokenResponse.accessToken,
+              accessTokenExpiration: tokenResponse
+                  .accessTokenExpirationDateTime?.millisecondsSinceEpoch,
+              refreshToken: tokenResponse.refreshToken,
+              shouldReconnect: 0);
+      Map? userInfo = await this.getUserInfo(apiAuthServiceAccountModel);
+      if (userInfo != null) {
+        apiAuthServiceAccountModel.displayName = userInfo['name'];
+        apiAuthServiceAccountModel.username = userInfo['id'];
+        apiAuthServiceAccountModel.email = userInfo['email'];
+        account = await this.upsert(apiAuthServiceAccountModel);
+        return account;
+      }
+    }
+    return null;
+  }
+
+  List<String> getProviders() {
+    return _apiAuthServiceRepository.providers!.keys
+        .map((el) => el as String)
+        .toList();
+  }
+
+  ApiAuthServiceProviderInterface? getProvider(
```

Comment:
Shouldn't we be returning the same instance of the service instead of a new one every time?